### PR TITLE
Data ingest

### PR DESCRIPTION
**Pull request by michaelkunc**
re-wrote the ingest scripts to load data to db.
 -added error handling
 -added a helper method to format the time field. 
 -added a kill and fill to ingest script. I.E. will delete records prior to loading.
 -changed ingest script to explicitly state file names in order to protect against a file getting accidentally   dumped into the csv_files directory.

added tests to for said ingest script.

changed models to allow for longer address lines.

### CODE REVIEW

**Comment by michaelkunc** [comment]
@worace & @SteveOfLA : could I ask both of you to review this? It was fairly beastly to untangle this one.

File: api/management/commands/csv_ingest.py [hunk 1]
```diff
@@ -1,21 +1,43 @@
...
+            try:
+                with open('csv_files/{}'.format(file), 'r') as f:
+                    reader = csv.reader(f)
+                    next(f)
+                    for r in reader:
+                        loc = Location(name=r[0], address_1=r[1], address_2=r[2], city=r[3], state=r[
+                                       4], zipcode=r[5], phone=r[6], latitude=r[7], longitude=r[8], category=r[9], website=r[10], active=True)
+                        loc.save()
+                        hours = list(filter(None, r[11:]))
+                        for i in [hours[i:i + 3] for i in range(0, len(hours), 3)]:
```
Comment (worace) [inline api/management/commands/csv_ingest.py]: I find this nested for comprehension a bit hard to follow, especially since it's not clear what the 3 pieces of the Hour data are when we start slicing them out.

I wonder if something like this might work as a way of labeling more clearly what those pieces are

```python
                        for i in range(0, len(hours), 3):
                            hour_components = hours[i:i + 3]
                            (day, open_time, close_time) = hour_components
                            hour = Hour(day=day,
                                     open_time=self._format_time(open_time),
                                     close_time=self._format_time(close_time),
                                     location=loc)
                            hour.save()
```

File: api/tests/command_test.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from django.core.management import call_command
```
Comment (worace) [inline api/tests/command_test.py]: I'm not totally sure this test is getting run when i run `make test` -- is there a special command you were using to run it? (I'm mostly basing this on how long the tests take to run which seems to quick for it to be running the full import task of 1220 records)

That also brings up another question which is whether we really want to run that full import on every test run. I wonder if it's possible to run for a slimmer set of data in the tests so it stays faster.

File: api/management/commands/csv_ingest.py [hunk 3]
```diff
@@ -1,21 +1,43 @@
...
+        csv_files = ['farmers-market.csv',
+                     'community_gardens.csv', 'food-pantry.csv', 'supermarket.csv']
+        location_count = 0
+        for file in csv_files:
+            try:
+                with open('csv_files/{}'.format(file), 'r') as f:
+                    reader = csv.reader(f)
+                    next(f)
+                    for r in reader:
+                        loc = Location(name=r[0], address_1=r[1], address_2=r[2], city=r[3], state=r[
```
Comment (michaelkunc) [inline api/management/commands/csv_ingest.py]: Fixed this. I think my auto-formatter went a little crazy. I agree it's easier to read when the attributes are stacked up.

File: api/management/commands/csv_ingest.py (hunk 1 above)
Comment (michaelkunc) [inline api/management/commands/csv_ingest.py]: slight variation on what you have here, but I think it captures the spirit of what you suggest.

File: api/tests/command_test.py (hunk 2 above)
Comment (michaelkunc) [inline api/tests/command_test.py]: I've been running this command with `python manage.py test` and I believe it is getting run. I can make the test fail by removing files or messing with the ingest command. 

I think it would be possible to run a slimmer data set in the future. While we're adding data sets, I'd like to keep testing the entire import.

Comment (worace) [inline api/tests/command_test.py]: good call i did not think to make the test break to see if it is working; the fact that it is running and the tests are still as fast as they are means the import is much faster than i was expecting, so my concern is probably unfounded


since it is so fast i agree with you we might as well leave the whole import in there in order to  have everything covered

**Comment by worace** [comment]
👍  looks good to me
